### PR TITLE
feat(testing): add playwright generator to convert from executors to plugin

### PR DESCRIPTION
**Pull request by Coly010**
## Current Behavior

The only way to migrate a project from playwright executor usage to playwright plugin usage is to do so manually.

## Expected Behavior

Add a generator that will convert projects from playwright executor usage to playwright plugin usage

### Usage

`nx g @nx/playwright:convert-to-inferred --project="myapp-e2e"` // Convert only `myapp-e2e` from executor to plugin usage
`nx g @nx/playwright:convert-to-inferred` // Migrate all projects

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/devkit/src/generators/plugin-migrations/plugin-migration-utils.ts [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+    tree: Tree,
+    projectRoot: string,
+    targetName: string,
+    context: CreateNodesContext
+  ) => CreateNodesResult | Promise<CreateNodesResult>,
+  executor: string,
+  createNodes: CreateNodes<T>,
+  projectOptionsTransformer: (
+    target: TargetConfiguration
+  ) => TargetConfiguration = (targetConfiguration) => targetConfiguration
```
Comment (FrozenPandaz) [inline packages/devkit/src/generators/plugin-migrations/plugin-migration-utils.ts:100]: Update this to your new version with `tree`, etc.

File: packages/devkit/src/generators/plugin-migrations/plugin-migration-utils.ts [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+      createdProject.targets[targetName];
+    delete createdTarget.command;
+    delete createdTarget.options?.cwd;
+
+    const projectConfig = readProjectConfiguration(tree, projectName);
+    let target = projectConfig.targets[targetName];
+
+    target = mergeTargetConfigurations(target, targetDefaultsForExecutor);
+    delete target.executor;
+    delete target.options?.config;
```
Comment (FrozenPandaz) [inline packages/devkit/src/generators/plugin-migrations/plugin-migration-utils.ts:150]: Move this to the `postTargetTransformer`

File: packages/playwright/src/generators/convert-to-inferred/lib/migrate-executor-to-plugin.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      tree,
+      '@nx/playwright/plugin',
+      include,
+      { targetName, ciTargetName: 'e2e-ci' }
+    );
+
+    for (const project of migratedProjects) {
+      allProjectsWithExecutor.delete(project);
+    }
+  }
```
Comment (FrozenPandaz) [inline packages/playwright/src/generators/convert-to-inferred/lib/migrate-executor-to-plugin.ts:64]: Move this into devkit.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: packages/playwright/src/generators/convert-to-inferred/convert-to-inferred.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  let project: ProjectConfiguration;
+  if (options.project) {
+    project = readProjectConfiguration(tree, options.project);
+  }
+
+  await migrateExecutorToPlugin(tree, project?.name, project?.root);
+
+  if (!options.skipFormat) {
+    await formatFiles(tree);
+  }
```
Comment (FrozenPandaz) [inline packages/playwright/src/generators/convert-to-inferred/convert-to-inferred.ts:35]: This should basically look like...

1. Get the graph
2. Call the devkit function
3. Format Files

File: packages/devkit/src/generators/plugin-migrations/plugin-migration-utils.ts [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+      workspaceRoot: tree.root,
+      nxJsonConfiguration,
+      configFiles,
+    };
+    const projectConfigurations = await createProjectsConfig(
+      tree,
+      projectFromGraph.data.root,
+      targetName,
+      context
+    );
```
Comment (FrozenPandaz) [inline packages/devkit/src/generators/plugin-migrations/plugin-migration-utils.ts:233]: This is weird... keep it for now... we should remove this ideally before we do too many plugins.

Let's get an initial version in.. and start using it.

File: packages/devkit/src/generators/plugin-migrations/plugin-migration-utils.ts [hunk 6]
```diff
@@ -0,0 +1,521 @@
...
+    return;
+  }
+
+  nxJson.plugins.push({
+    plugin: pluginPath,
+    include,
+    options: pluginOptions,
+  });
+
+  updateNxJson(tree, nxJson);
```
Comment (FrozenPandaz) [inline packages/devkit/src/generators/plugin-migrations/plugin-migration-utils.ts:353]: Let's change this to update the existing plugin to update includes.

At the end check again if include includes every configfile... if it does... we can then remove the include property.

We will need update the playwright init.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

**Review by isaacplmann — APPROVED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**